rfcs/rfc-2-font-measuring.md

```diff
@@ -10,49 +10,54 @@ I propose two different solutions for the issues above.
 1. KayakContext keeps track of all fonts that are loaded. This allows us to use the available measuring to properly size the Text widget.
 2. bevy_kayak_ui uses include_bytes to load in a default font(probably Roboto).
 
-For the first solution an example of how that would look is something like:
+
+## How this will look
+
+We'll need a few new types to store the fonts. I would also like to make something more generic and reusable in the future. It'll look something like this:
+
+`AssetHandle`:
 ```rust
-fonts: HashMap<u16, KayakFont>,
+pub struct AssetHandle<T> {
+    id: u32, // Optionally UUID here although I don't think its required..
+    phantom_data: PhantomData<T>,
+}
 ```
 
-For the second issue an example would look like:
+`AssetStorage`:
 ```rust
-let image = Image::new(
-    size,
-    dimension: TextureDimension::D2,
-    data: include_bytes!("./assets/roboto.png"),
-    format: TextureFormat::Rgba8Unorm,
-);
-let atlas_handle = images_assets.add(image);
-
-let mut font = KayakFont::new(
-    Sdf::from_bytes(include_bytes!("./assets/roboto.kayak_font")),
-    atlas_handle,
-);
+pub struct AssetStorage<T> {
+    assets: HashMap<AssetHandle<T>, T>,
+    // Stores bindings that notify us when changes are made.
+    // Optionally we can make Binding Hash here. Not sure if I like that though..
+    binding: HashMap<AssetHandle<T>, Binding<AssetHandle<T>>>,
+}
 ```
 
-## Additional issues:
-The text widget will need to be re-rendered when it's font is loaded in. I suggest quite a simple way of doing this:
+`KayakContext`:
 ```rust
-fonts: Binding<HashMap<u16, KayakFont>>,
+pub struct KayakContext {
+    ..
+    // Stores AssetStorage generically.
+    assets: Resources,
+}
+
+impl KayakContext {
+    pub fn get_asset<T>(&self, asset_handle: AssetHandle<T>) -> &T {
+        // Throw error if AssetStorage doesn't exist in the asset resources.
+        ..
+    }
```

Comment:
I feel like this should return a `Result` instead. Then we can create an asset-related error type that makes handling the error a bit easier.

~~Also, should this return a `&Binding<T>`?~~ Actually, it probably shouldn't since they can't bind to it. What would be the method of attaching/removing a listener for an asset on a given widget?

Reply:
Sorry you are right it probably should be `Binding<T>`. I think I mixed some things up.

Reply:
And this method will automatically bind the widget to the `Binding<T>` then?